spanner/api/SpannerTest/Tests.cs

```diff
@@ -558,6 +558,22 @@ void TestCreateDatabase()
             Assert.Contains("Singers", output.Stdout);
         }
 
+        [Fact]
+        void TestQueryOptions()

```

Comment:
If you remove the sync methods on the sample projects, then you'll have to make these async Task TestQueryOptions()...

Reply:
Done but I'm not sure that I'm doing this right.

I wanted to add an `await` on `_spannerCmd.Run() ` but that gives me the following build error:
```
Tests.cs(564,36): error CS1061: 'ConsoleOutput' does not contain a definition for 'GetAwaiter' and no accessible extension method 'GetAwaiter' accepting a first argument of type 'ConsoleOutput' could be found (are you missing a using directive or an assembly reference?) [/google-cloud-dotnet/spanner/api/SpannerTest/SpannerTest.csproj]
```

If I don't add the `await` it gives a warning:
```
Tests.cs(562,20): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/google-cloud-dotnet/spanner/api/SpannerTest/SpannerTest.csproj]
```

Reply:
I seem to have failed at doing this correctly because the async tests couldn't be run with the existing setup and I didn't know how to fix it.
```
  X GoogleCloudSamples.Spanner.SpannerTests.TestRunCommandWithQueryOptions [31ms]
  Error Message:
   System.InvalidCastException : Unable to cast object of type 'AsyncStateMachineBox`1[System.Threading.Tasks.VoidTaskResult,GoogleCloudSamples.Spanner.Program+<RunCommandWithQueryOptionsAsync>d__121]' to type 'System.Int32'.
  Stack Trace:
     at GoogleCloudSamples.Spanner.Program.Main(String[] args) in /google-cloud-dotnet/spanner/api/Spanner/Program.cs:line 3753
   at GoogleCloudSamples.CommandLineRunner.Run(String[] arguments) in /google-cloud-dotnet/testutil/TestUtil.cs:line 135
   at GoogleCloudSamples.Spanner.SpannerTests.TestRunCommandWithQueryOptions() in /google-cloud-dotnet/spanner/api/SpannerTest/Tests.cs:line 573
```

In the end I set up the async method in the same way as `ReadStaleDataAsync()` but that means the test itself is not async.

Reply:
As discussed, I reverted back to using the sync method.